### ISSUE TITLE
Order of tags is non-deterministic

### ISSUE THREAD

**Issue by neon-dev** [smallrye/smallrye-open-api#117]
I'm not sure why, but running the same project on different systems (OSes) yields different tag orders and therefore different endpoint group order in Swagger UI.

### Example
Three classes, all in the same package:
```java
@Tag(name = "A")
public class A {
  /* endpoints here */
}
```
```java
@Tag(name = "B")
public class B {
  /* endpoints here */
}
```
```java
@Tag(name = "C")
public class C {
  /* endpoints here */
}
```

### Results
Windows system:
```yml
---
openapi: 3.0.1
info:
  title: test
  version: "1.0"
tags:
- name: A
- name: B
- name: C
```

Linux system:
```yml
---
openapi: 3.0.1
info:
  title: test
  version: "1.0"
tags:
- name: C
- name: A
- name: B
```

**Comment by EricWittmann** [comment]
I guess this must depend on the order in which the classes are processed... I am not sure offhand why that would be different depending on the system.  But I can imagine.

What might we do or even expect about this?

**Comment by neon-dev** [comment]
Since even paths (and request/response-fields, though not intuitive see #87) are alphabetically sorted, I'd expect tags also to be ordered this way.

**Comment by EricWittmann** [comment]
Ah ok.  Sorting them is certainly doable!

**Comment by neon-dev** [comment]
I guess there's no simpler alternative to get consistent results.
It was quite irritating seeing this difference between the two systems after just changing a class name and it's tag. Before that everything was identical, so I'm not even sure if my example above would reproduce the issue. Because my actual classes and tag names are obviously not A, B and C.

Surely some folks would like to have control over the order, but then we shouldn't only consider tags for such a feature and honestly it's not that important. The worst problem for me currently is the issue I linked above (left another comment there).

**Comment by msavy** [comment]
Class file ordering of most elements is non-deterministic (especially since Java 7 and 8 -- previously in practice it was more stable, despite not guaranteeing it), due to hashing seeds, vagaries of compiler, and so forth. Just to be clear @neon-dev, I assume you find this annoying rather than causing technical issues?

JLS Chapter 13 describes this, if you're interested.

The order you declare stuff in the java source is not guaranteed in the compiled class-file.

**Comment by neon-dev** [comment]
Yes, no technical issues. Just unexpected to see non-identical Swagger-UI lists for the exact same build version on different systems.
From user standpoint there should be some kind of consistent ordering. Be it by ordering classes by their fully qualified names before parsing, or simple alphabetical tag sorting afterwards. I'd be fine with both, since I don't care that much how exactly the order looks, just that it's consistent and comprehensible.

**Comment by MikeEdgar** [comment]
I'm curious what the group's thoughts are on this one. Should there be a general approach to ordering top-level object keys (i.e. tags, sub-items in components) as is done with paths? Any considerations with how the scanned items are merged with the static file items? I would expect the ordering to apply across all sources if done at all. My personal thought is that ordered entries would be "nice to have" and make a spec more readable (to a human).

**Comment by EricWittmann** [comment]
I think sorting/ordering for **certain** elements by default may be desirable.  Tags (in this use-case at least) seems to make sense.  I'm trying to work through whether ordering/sorting is always a good idea or not.  Hard to say without more use-cases documented.  

I do think that if there is a static file defined, then any ordering in the static file should be maintained.

**Comment by MikeEdgar** [comment]
I'm mainly thinking about only tags and the maps directly under components. Rethinking what I said about the static file, you're right that the order there should take precedence.
